app/script/notification/NotificationRepository.js

```diff
@@ -564,31 +564,19 @@ z.notification.NotificationRepository = class NotificationRepository {
   _createTrigger(messageEntity, connectionEntity, conversationEntity) {
     const conversationId = this._getConversationId(connectionEntity, conversationEntity);
 
-    if (messageEntity.is_content() && messageEntity.isSelfMentioned()) {
+    const containsSelfMention = messageEntity.is_content() && messageEntity.isSelfMentioned();
+    if (containsSelfMention) {
       return () => amplify.publish(z.event.WebApp.CONVERSATION.SHOW, conversationEntity, messageEntity, true);
     }
 
-    if (!messageEntity.is_member()) {
-      return () => amplify.publish(z.event.WebApp.CONVERSATION.SHOW, conversationEntity || conversationId);
+    const isConnectionRequest = messageEntity.is_member() && messageEntity.isConnectionRequest();
+    if (isConnectionRequest) {
+      return () => {
+        amplify.publish(z.event.WebApp.CONTENT.SWITCH, z.viewModel.ContentViewModel.STATE.CONNECTION_REQUESTS);
+      };
     }
 
-    switch (messageEntity.memberMessageType) {
-      case z.message.SystemMessageType.CONNECTION_ACCEPTED:
-      case z.message.SystemMessageType.CONNECTION_CONNECTED: {
-        return () => amplify.publish(z.event.WebApp.CONVERSATION.SHOW, conversationId);
-      }
-
-      case z.message.SystemMessageType.CONNECTION_REQUEST: {
-        return () => {
-          amplify.publish(z.event.WebApp.CONTENT.SWITCH, z.viewModel.ContentViewModel.STATE.CONNECTION_REQUESTS);
-        };
-      }
-
-      default: {
-        const message = `No notification trigger for message '${messageEntity.id} in '${conversationId}'.`;
-        this.logger.log(this.logger.levels.OFF, message);
-      }
-    }
+    return () => amplify.publish(z.event.WebApp.CONVERSATION.SHOW, conversationEntity || conversationId);
```

Comment:
Now that we removed the whitelist, we might run into publishing `CONVERSATION.SHOW` events for things we don't want. Is that expected?

Reply:
Yes. That is actually the bug. Check https://github.com/wireapp/wire-webapp/pull/4595. My PR from two days ago refactored some code and I missed the fact that the switch case had a fall through. The conversation show was the default, despite some exceptions doing already the exact same thing. This simplifies the code a lot and fixes a bug from the original code that was masked by the fallthough.

Reply:
Ok then, if you feel it's safe, then LGTM. thanks for the explanation.

Reply:
Testing the trigger in the unit tests seems complicated. Maybe we should add that at some point. For now we were only looking at the stringified content.